File: fabric-apollo/fabric-apollo-amqp-protocol/src/main/scala/org/fusesource/fabric/apollo/amqp/protocol/AmqpCommandHandler.scala
/**
 * Copyright (C) 2010-2011, FuseSource Corp.  All rights reserved.
 *
 *     http://fusesource.com
 *
 * The software in this package is published under the terms of the
 * CDDL license a copy of which has been included with this distribution
 * in the license.txt file.
 */

package org.fusesource.fabric.apollo.amqp.protocol

import org.fusesource.fabric.apollo.amqp.codec.AmqpHandler
import org.fusesource.fabric.apollo.amqp.codec.AmqpProtocolHeader
import org.fusesource.fabric.apollo.amqp.codec.types._

/**
 *
 */
class AmqpCommandHandler(connection:ConnectionHandler, s:AmqpSession) extends AmqpHandler {

  val session = Option(s)
  def handleEmpty: Unit = {}
  def handleUnknown(body: AmqpType[_, _]): Unit = {}

  def handleProtocolHeader(protocolHeader: AmqpProtocolHeader) = connection.header(protocolHeader)
  def handleClose(close: AmqpClose) = connection.close
  def handleOpen(open: AmqpOpen) = connection.open(open)
  def handleSaslChallenge(saslChallenge: AmqpSaslChallenge) = connection.handleSaslChallenge(saslChallenge)
  def handleSaslResponse(saslResponse: AmqpSaslResponse) = connection.handleSaslResponse(saslResponse)
  def handleSaslMechanisms(saslMechanisms: AmqpSaslMechanisms) = connection.handleSaslMechanisms(saslMechanisms)
  def handleSaslInit(saslInit: AmqpSaslInit) = connection.handleSaslInit(saslInit)
  def handleSaslOutcome(saslOutcome: AmqpSaslOutcome) = connection.handleSaslOutcome(saslOutcome)

  def handleFlow(flow: AmqpFlow) = session.foreach((s) => s.flow(flow))
  def handleTransfer(transfer: AmqpTransfer) = session.foreach((s) => s.transfer(transfer))
  def handleDetach(detach: AmqpDetach) = session.foreach((s) => s.detach(detach))
  def handleDisposition(disposition: AmqpDisposition) = session.foreach((s) => s.disposition(disposition))
  def handleEnd(end: AmqpEnd) = session.foreach((s) => s.end(end))
  def handleBegin(begin: AmqpBegin) = session.foreach((s) => s.begin(begin))
  def handleAttach(attach: AmqpAttach) = session.foreach((s) => s.attach(attach))

}

File: fabric-apollo/fabric-apollo-amqp/fabric-apollo-amqp-codec2/src/main/java/org/fusesource/fabric/apollo/amqp/codec/AmqpCommandHandler.java
/**
 * Copyright (C) 2010-2011, FuseSource Corp.  All rights reserved.
 *
 *     http://fusesource.com
 *
 * The software in this package is published under the terms of the
 * CDDL license a copy of which has been included with this distribution
 * in the license.txt file.
 */

package org.fusesource.fabric.apollo.amqp.codec;

import org.fusesource.fabric.apollo.amqp.codec.types.*;

public interface AmqpCommandHandler {

    public void handleFlow(Flow flow) throws Exception;

    public void handleClose(Close close) throws Exception;

    public void handleOpen(Open open) throws Exception;

    public void handleSASLOutcome(SASLOutcome saslOutcome) throws Exception;

    public void handleTransfer(Transfer transfer) throws Exception;

    public void handleDetach(Detach detach) throws Exception;

    public void handleSASLInit(SASLInit saslInit) throws Exception;

    public void handleSASLMechanisms(SASLMechanisms saslMechanisms) throws Exception;

    public void handleSASLResponse(SASLResponse saslResponse) throws Exception;

    public void handleDisposition(Disposition disposition) throws Exception;

    public void handleEnd(End end) throws Exception;

    public void handleBegin(Begin begin) throws Exception;

    public void handleSASLChallenge(SASLChallenge saslChallenge) throws Exception;

    public void handleAttach(Attach attach) throws Exception;
}